src/util/thread_runner.h

```diff
@@ -164,56 +172,14 @@ class ThreadRunner : public ThreadExecutor {
         }
     }
 
-protected:
-    std::shared_ptr<Config> config;
-
-    /// \brief thread method is injected
-    void threadProc() override
-    {
-        targetProc(target);
-        log_debug("ThreadRunner: Terminating {}", threadName);
-    }
-
-    /// \brief start the thread
-    void startThread() override
-    {
-#if !defined(SOLARIS) && !defined(__HAIKU__)
-        // default scoping on Solaroid systems is in fact PTHREAD_SCOPE_SYSTEM
-        // plus, setting PTHREAD_EXPLICIT_SCHED requires elevated privileges
-        // while Haiku doesn't implement pthread_attr_setinheritsched yet
-        if (config->getBoolOption(CFG_THREAD_SCOPE_SYSTEM)) {
```

Comment:
You can also drop the option from config

Reply:
Good spot!